src/app/hooks/useUserRights.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { client } from '../lib/http-client';
+import { configuration } from '../lib/configuration';
+import {
+  RightToOrganization,
+  UserRights,
+  AUTH_SERVICE,
+  RIGHT,
+  getOrgsForRight,
+} from '../lib/auth';
+
+export const getUserRights = async (): Promise<UserRights> => {
+  const response = await client.get(configuration.kayttoikeusUrl);
+  if (!response.data) {
+    throw Error('Unable to get user data');
```

Comment:
Missä tilanteessa tämä tapahtuu, että awaitista ei tule virhettä, mutta data on tyhjä? Jos tämä on ihan validi keissi, niin pitäskö tästä heittää jokin spesifimpi virhe, jonka voisi esittää error boundaryssa paremmin?

Reply:
Voisin oikeastaan poistaa tämän. Tuosta kuitenkin lentää poikkeus jos näin käy.